toolkit/pallets/block-production-log/src/lib.rs

```diff
@@ -96,5 +112,15 @@ pub mod pallet {
 			Log::<T>::put(to_retain);
 			to_return
 		}
+
+		pub fn peek_prefix<'a>(slot: Slot) -> impl Iterator<Item = (Slot, T::BlockProducerId)> {
```

Comment:
lifetime is not needed 
```suggestion
		pub fn peek_prefix(slot: Slot) -> impl Iterator<Item = (Slot, T::BlockProducerId)> {
```